src/perf/best-practices.md

```diff
@@ -0,0 +1,375 @@
+screen is built at startup time.
+
+For more information and examples, check out:
+
+* [Working with long lists][] in the [Cookbook][]
+* [Creating a `ListView` that loads one page at a time][]
+  a community article by AbdulRahman AlHamali
+* [`Listview.builder`][] API
+
+[Cookbook]: {{site.url}}/cookbook
+[Creating a `ListView` that loads one page at a time]: {{site.medium}}/saugo360/flutter-creating-a-listview-that-loads-one-page-at-a-time-c5c91b6fabd3
+[`Listview.builder`]: {{site.api}}/flutter/widgets/ListView/ListView.builder.html
+[Working with long lists]: {{site.url}}/cookbook/lists/long-lists
+
+<a href="minimize-layout-passes-caused-by-intrinsic-operations"></a>
+### Minimize layout passes caused by intrinsic operations
+
+If you've done much Flutter programming, you are
+probably familiar with [how layout and constraints work][]
+when creating your UI and you might have memorized Flutter's
+basic layout rule: **Constraints go down. Sizes go up.
+Parent sets position.**
+
+For some widgets, particularly grids and lists,
+the layout process can be expensive.
+For example, consider a 10x500 grid of `Card`s.
+A grid should have uniformly sized cells,
+so the layout code performs a pass,
+starting from the root of the grid (in the widget tree),
+asking each card in the visible grid to return
+its _intrinsic_ size&mdash;the size
+that the widget prefers, assuming no constraints.
+With this information,
+the framework calculates a uniform cell size,
+and re-visits the grid objects,
+_telling_ each card what size to use. 
+
+If your code performs unnecessary layout passes,
+it can slow performance. To determine if this
+is your issue, enable the **Track layouts option**
+in DevTools (disabled by default),
+and look at the app's [stack trace][]
+to learn how many layout passes were performed.
+[PENDING: Where is the track layout option in DevTools?]
+
+[PENDING: How does once determine "excessive"?]
+[PENDING: Is the primary way to fix this by lazy loading
+the grid or list? Or is there something else?]
+
+To dive even deeper into how layout works,
+check out the [layout and rendering][]
+section in the [Flutter architectural overview][].
+
+
+[Flutter architectural overview]: {{site.url}}/resources/architectural-overview
+[how layout and constraints work]: {{site.url}}/development/ui/layout/constraints
+[layout and rendering]: {{site.url}}/resources/architectural-overview#layout-and-rendering
+[stack trace]: {{site.url}}/development/tools/devtools/cpu-profiler#flame-chart
+
+---
+
+##  Build and display frames in 16ms
+
+Since there are two separate threads for building
+and rendering, you have 16ms for building,
+and 16ms for rendering on a 60Hz display.
+If latency is a concern,
+build and display a frame in 16ms _or less_.
+Note that means built in 8ms or less,
+and rendered in 8ms or less,
+for a total of 16ms or less.
+If missing frames (jankiness) is a concern,
+then 16ms for each of
+the build and render stages is OK.
```

Comment:
I am confused by what this paragraph is trying to tell me. Do I hav 16ms for each of render or build or 16ms for both together?

Reply:
This is an old paragraph. I believe it's saying 16ms for both together. Yes?

Reply:
I deleted the confusing sentence.